AutoCollection/HttpDependencies.ts

```diff
@@ -63,6 +64,11 @@ class AutoCollectHttpDependencies {
             var shouldCollect = !(<any>options)[AutoCollectHttpDependencies.disableCollectionRequestOption] &&
                 !(<any>request)[AutoCollectHttpDependencies.alreadyAutoCollectedFlag];
 
+            // If someone else patched traceparent headers onto this request
+            if (options.headers && options.headers['user-agent'] && options.headers['user-agent'].toString().indexOf('azsdk-js') !== -1) {
```

Comment:
Can you explain what is going on here?, what is azsdk-js is and why ignoring it?

Reply:
I should update the comment here because its not 100% clear.

Azure SDK patches outgoing http calls made via their SDK (e.g. GET a storage account blob). We patch ALL outgoing http calls, which intersects with azure SDK patching, so we need some way to prevent this double telemetry/patching. Another possibility, which I think Java SDK does, is to check for presence of `traceparent` header in the outgoing http call. The user will never set this header themselves, it is only set by APM SDKs. However, this would break patching if another SDK (e.g. opentelemetry) is used. Once this SDK supports opentelemetry patching, then this could be switched to check for traceparent header.  Until then, azure sdk sets this user-agent field on all of their outgoing http headers, so I think its okay to just check for this for now.

Summary:
1. Azure SDK patches their outgoing http calls to add `traceparent` and `user-agent`
2. This SDK can see the headers of an outgoing http call before it gets sent out, so we can not track the outgoing http call if Azure SDK tracer will already track it.
3. If I just look for `traceparent` and user instruments with multiple W3C compatible SDKs, then our SDK won't track, even though it should. Should only step-back for Azure SDK